lib/connections/service.go

```diff
@@ -701,6 +701,10 @@ func (s *service) ConnectionStatus() map[string]ConnectionStatusEntry {
 }
 
 func (s *service) setConnectionStatus(address string, err error) {
+	if errors.Cause(err) != context.Canceled {
+		return
+	}
```

Comment:
Isn't this error condition inverted? Why do we want to only save error info for `context.Canceled`?